### PR TITLE
Fixing commands

### PR DESCRIPTION
**Pull request by emeraldarcher**
Got the ListAI Command working on the client/server.

### CODE REVIEW

File: java/src/client/services/ServerProxy.java [hunk 1]
```diff
@@ -312,7 +315,21 @@ public void addAI(String aiType) {
...
-        return null;
+        String result;
+
+        try {
+            result = Utils.sendPost(url, null);
+            assert result != null;
+        } catch (BadHttpRequestException e) {
+            e.printStackTrace();
+            this.showMessageViewForHttpError(e.getMessage());
+            return null;
```
Comment (derekargueta) [inline java/src/client/services/ServerProxy.java]: Have you tested how the client responds when the returned list *is* null?

File: java/src/server/commands/game/ListAICommand.java [hunk 2]
```diff
@@ -22,12 +24,17 @@
      */
     @Override
     public CommandExecutionResult execute() {
-        return null;
+        try {
+            return Config.facade.listAI(0);
+        } catch (ListAIException e) {
+            e.printStackTrace();
+            return new CommandExecutionResult("Error listing AI types!");
```
Comment (derekargueta) [inline java/src/server/commands/game/ListAICommand.java]: If an error occurs, you need to call `.triggerError(int)` on the `CommandExecutionResult` object, passing in a status code that will be returned to the client. Otherwise the client will receive a 200.

File: java/src/shared/model/ai/AIType.java [hunk 3]
```diff
@@ -4,5 +4,6 @@
  * Created by Kyle 'TMD' Cornelison on 3/19/2016.
  */
 public enum AIType {
-    LARGEST_ARMY
+    LARGEST_ARMY,
+    RODHAMMER
```
Comment (derekargueta) [inline java/src/shared/model/ai/AIType.java]: What is the purpose of this change? Why are we adding this AI type?

**Comment by derekargueta** [comment]
@dannyharding10 @yeyyosing please review

File: java/src/shared/model/ai/AIType.java (hunk 3 above)
Comment (emeraldarcher) [inline java/src/shared/model/ai/AIType.java]: We can remove it, I was testing what happens with multiple AIs just to make sure

File: java/src/client/services/ServerProxy.java (hunk 1 above)
Comment (emeraldarcher) [inline java/src/client/services/ServerProxy.java]: Fixed

File: java/src/server/commands/game/ListAICommand.java (hunk 2 above)
Comment (emeraldarcher) [inline java/src/server/commands/game/ListAICommand.java]: Fixed

File: java/src/shared/model/ai/AIType.java (hunk 3 above)
Comment (emeraldarcher) [inline java/src/shared/model/ai/AIType.java]: removed but added a todo for phase 4

**Comment by emeraldarcher** [comment]
Can you guys review this now?

**Comment by derekargueta** [comment]
@emeraldarcher can you also add unit tests to this PR for the new functionality? Unit tests are a nice way to quickly validate that your code works without having to run the server and manually hit it.
